apps/vanilla/src/pages/black-jack/features/dice-generator.ts

```diff
@@ -0,0 +1,48 @@
+import { getRandomIntegerNumber } from '@js-camp/core/utils/get-integer-random';
+
+import { Subscriber } from '../models/subscriber';
+
+import { Publisher } from '../models/publisher';
+
+import { PlayerTurnResult } from '../models/player-turn-result';
+
+import { Player } from './player';
+
+/** Dice generator. */
+export class DiceGenerator extends Publisher<PlayerTurnResult> implements Subscriber<number> {
+	public constructor(private readonly sidesCount: number) {
+		super();
+	}
+
+	/**
+	 * Notify when a dice turn is generated.
+	 * @param playerTurnResult Turn and dice result of player.
+	 **/
+	public override notify(playerTurnResult: PlayerTurnResult): void {
+		const currentPlayer = (this.subscriberArr as Player[]).find(s => s.index === playerTurnResult.playerIndex);
+		if (currentPlayer !== undefined) {
+			currentPlayer.update(playerTurnResult);
+		}
```

Comment:
As I said before, let's use non-strict comparation for `null` and `undefined` value.
```suggestion
		if (currentPlayer != null) {
			currentPlayer.update(playerTurnResult);
		}
```